### DISCUSSION TITLE
How to install plugin at specific commit

### DISCUSSION THREAD

**Discussion by pianocomposer321** [zdharma-continuum/zinit#193] [Q&A] [answered]
Hey,

How do I install a plugin at a specific point in the repo's hostory, like a specific branch or commit hash? There's a plugin I'm trying out, but there's a bug in the most recent version, so I want to try a version before the bug was introduced. How would I go about doing this?

Thanks in advance!

**Comment by vladdoster** [comment, marked as answer]
@pianocomposer321 This is documented in the docs.

https://github.com/zdharma-continuum/zinit#cloning-options

**Comment by vladdoster** [comment]
@pianocomposer321 If the link I posted resolves your question, please mark as the `answer` for people who might stumble upon this later.

**Reply by pianocomposer321** [reply]
Unfortunately, I don't have time to test this right now. I removed the plugin from zsh a while ago, because the bug was causing too many problems and I didn't have a way to pin a commit (hence this question). When I get a chance to try this out, I will get back with you.

I will say though, assuming the option you're referring to is the "ver" keyword, if this is the solution, it might be a good idea to make it a bit more discoverable. It's not even clear from the description that this option is for regular plugins, as it opens by talking about binary releases. It seems to me that for common options it may be beneficial to have a list showing what one might want to do and the options that will accomplish it, rather than just having a list of options with a description. Doing only the latter makes it difficult to find the right option, because new users won't know the name of the option they're looking for, just what thing they're trying to accomplish.

**Comment by vladdoster** [comment]
> Used with from"gh-r" (i.e., downloading a binary release, e.g., for use with as"program") – selects which version to download. Default is latest, can also be explicit `ver"latest`. Works with regular plugins and packages (pack ice) checkouts (e.g., ver"branch" for a specific version). Does not work with snippets.

I agree the documentation in its current state is hard to navigate, scattered across multiple places, and needs a lot of improvement. However, the `ver` ice description states it works with regular plugins.

Your insight and suggested improvements are noted. I'll follow up with you once we have new docs and see what you think :).

**Reply by pianocomposer321** [reply]
You're right, when I said "is not clear" I should have said "is not immediately clear". It says "Used with from"gh-r"", which seems to immediately indicate that this is the only thing that is option is for. It would be easy for a user to read this first sentence and immediately stop, thinking that this is not the correct option because they are using from"github". (I know, because this is precisely what I did 😉).

Thank you for your help though, I will definitely try this out when I get a chance.

**Reply by vladdoster** [reply]
@pianocomposer321 Did my suggestion solve your issue?
